File: src/my_project/medium_problems/from51to100/longest_increasing_subsequence_dp.py
from typing import List 

class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        
        if not nums: 
            return 0 
        
        len_nums: int = len(nums)
        dp = [1]*len_nums
        
        for i in range(1,len_nums):
            for j in range(i):
                if nums[i] > nums[j]:
                    dp[i] = max(dp[i], 1+dp[j])
                
        
        return max(dp)

File: my_project_typescript/medium_problems_typescript/from1to50/longestIncreasingSubsequenceDP.ts
function lengthOfLIS(nums: number[]): number {

    if (!nums){
        return 0
    }

    let lenNums: number = nums.length

    let dp: Array<number> = Array(lenNums).fill(1)


    for (let i = 1; i < lenNums; i++){
        for (let j = 0; j < i; j++){
            if (nums[i] > nums[j]){
                dp[i] = Math.max(dp[i], 1+dp[j])
            }
        }
    }


    return dp.reduce( (a,b) => Math.max(a,b),0)
};

console.log(lengthOfLIS([10,9,2,5,3,7,101,18]))